src/lib/Libattr/attr_fn_arst.c

```diff
@@ -957,19 +957,13 @@ encode_arst_bs(attribute *attr, pbs_list_head *phead, char *atname, char *rsname
 
 
 	i = (int)(attr->at_val.at_arst->as_next - attr->at_val.at_arst->as_buf);
-	if (mode == ATR_ENCODE_SAVE) {
-		separator = '\n';	/* new-line for encode_acl  */
-		/* allow one extra byte for final new-line */
-		j = i + 1;
-	} else {
-		separator = ',';	/* normally a comma is the separator */
-		j = i;
-	}
+	separator = ',';	/* normally a comma is the separator */
```

Comment:
On the server side, we won't have tihis problem anymore since this ENCODE_SAVE mode only applies when saving/retrieving data from flat files. Now we use database so it won't be an issue. The only one that still use flat file is pbs_mom, and our instruction for upgrade is to stop scheduling, requeue all jobs, before killing the old version of PBS, and starting in a new one. We won't then run into problems with old version  encoding one way and new version mom reading another way. A site can run into problem if they don't follow our upgrade instruction and would need to do a 'kill -9' of old pbs_mom, and start the new pbs_mom  running mom hooks which would corrupt the environment.

Reply:
Are you referring to upgrade instructions for PBS Pro OSS? I wasn't aware we had instructions for upgrading.

Reply:
Thinking about this more, don't think there's an upgrade instruction for PBS Pro in OSS. I think the normal process is to uninstall previous version of PBS completely, leaving PBS_HOME, then install a new version.  These steps should not result in a situation where old version of mom encoded job Variable_List one way, and a new version of mom decoded it another way.

Reply:
Upgrades will most likely use "rpm -U" which should do nothing more than replace the existing binaries and update PBS_EXEC in pbs.conf if necessary (unlikely). It will not restart any PBS services, so any mom running prior to the upgrade will continue to run until the machine is rebooted or PBS restarted manually. I think there is a relatively low risk of encountering an incompatibility, but I wanted to check with you. What do you think?

Reply:
I agree, it's relatively low risk.